python_homework/iter_practice/practicer_iter.py

```diff
@@ -0,0 +1,244 @@
+        Args:
+            self
+            sequence: amino acid sequence for changes.
+        Returns:
+            new_seq: string with altered amino acid sequence.
+        """
+        new_seq = []
+        changes = [self.make_deletion, self.make_substitution, self.make_insertion]
+
+        for i in sequence:
+            if random.randint(0, 1):
+                new_seq.append(i)
+
+            else:
+                selected_change = random.choice(changes)
+                new_seq.append(selected_change(i))
+
+        new_seq = ''.join(new_seq)
+        return new_seq
+
+    def make_deletion(self, letter):
+        """Remove an amino acid from a sequence.
+        Return empty string for this.
+        Args:
+            self
+            letter: letter for changes.
+        Returns:
+            '': empty string.
+        """
+        return ''
+
+    def make_substitution(self, letter):
+        """Replaces an amino acid in a sequence with a random amino acid.
+        Return random letter for this.
+        Args:
+            self
+            letter: letter for changes.
+        Returns:
+            string: randomly selected letter from AA_ALPHABET.
+        """
+        return random.choice(self.AA_ALPHABET)
+
+    def make_insertion(self, letter):
+        """Insert a random amino acid in a sequence.
+        Return random letter and initial letter for this.
+        Args:
+            self
+            letter: letter for changes.
+        Returns:
+            string: initial letter plus randomly selected letter from AA_ALPHABET.
+        """
+        return ''.join([letter, random.choice(self.AA_ALPHABET)])
+
+    def read_sequences(self, file_path):
+        """Takes as input the path to the FASTA file and outputs list with sequence.
+        Args:
+            file_path: Path to the FASTA file.
+        Returns:
+            list with sequences strings.
+        """
+        temp_list = []
+        seq_strings = []
+
+        with open(file_path) as inf:
+            for line in inf:
+                line = line.rstrip()
+
+                if line.startswith('>'):
+                    if temp_list:
+                        seq_strings.append(''.join(temp_list[1:]))
+                        temp_list = []
+
+                temp_list.append(line)
+
+        return seq_strings
+
+
+def fasta_reader(file_path):
+    """Generetor.
+    Takes as input the path to the FASTA file and outputs pairs sequence id and sequence in turn.
+    Args:
+        file_path: Path to the FASTA file.
+    Yields:
+        seq_id, sequence.
+    """
+    temp_list = []
+
+    with open(file_path) as inf:
+        for line in inf:
+            line = line.rstrip()
+
+            if line.startswith('>'):
+                if temp_list:
+                    yield temp_list[0], ''.join(temp_list[1:])
+                    temp_list = []
+
+            temp_list.append(line)
+
```

Comment:
У тебя не считывается последняя последовательность из файла, так как `yield` отдаёт только то, что было до ">". Чтобы вернуть последнюю последовательность можно сделать отдельный `yield` в самом конце
```suggestion
yield temp_list[0], ''.join(temp_list[1:])
```